### PR TITLE
[SYCLomatic] Enable migration of atomicDec

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/APINames.inc [hunk 1]
```diff
@@ -361,7 +361,7 @@ ENTRY(atomicExch, atomicExch, true, NO_FLAG, P0, "Successful")
 ENTRY(atomicMin, atomicMin, true, NO_FLAG, P0, "Successful")
 ENTRY(atomicMax, atomicMax, true, NO_FLAG, P0, "Successful")
 ENTRY(atomicInc, atomicInc, true, NO_FLAG, P0, "Successful: helper function")
-ENTRY(atomicDec, atomicDec, false, NO_FLAG, P0, "no direct mapping yet")
+ENTRY(atomicDec, atomicDec, true, NO_FLAG, P0, "Successful: helper function")
```
Comment (wenhuiNi) [inline clang/lib/DPCT/APINames.inc]: "Successful: helper function", Just a ask, should be filled in "Successful: helper function" in the future when using help function as API mapping?

Comment (wenhuiNi) [reply clang/lib/DPCT/APINames.inc]: I think we need a consistent format of MigrationDesc

**Review by wenhuiNi — APPROVED** [review]

File: clang/runtime/dpct-rt/include/atomic.hpp.inc [hunk 2]
```diff
@@ -383,6 +383,37 @@ inline T atomic_fetch_max(T *addr, T operand,
 }
 // DPCT_LABEL_END
 
+// DPCT_LABEL_BEGIN|atomic_fetch_compare_dec|dpct
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+/// Atomically decrease the value stored in \p addr if old value stored in \p addr
+/// is equal to zero or greater than \p operand, else set 0 to the value stored in \p addr.
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/atomic.hpp.inc]: the commnet is not correct.  Doesn't align with the code.

Comment (tomflinda) [reply clang/runtime/dpct-rt/include/atomic.hpp.inc]: Will align it in the updated PR.

File: clang/lib/DPCT/APINames.inc (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/APINames.inc]: Will update them with "Successful".

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**